### PR TITLE
Add --tier option to provide default settings

### PR DESCRIPTION
**Pull request by jemrobinson**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).

### :vertical_traffic_light: Depends on

n/a

### :arrow_heading_up: Summary

Add --tier option to provide default settings

### :closed_umbrella: Related issues

n/a

### :microscope: Tests

Tested on new config generation

### CODE REVIEW

**Review by JimMadge — COMMENTED** [review]
We should have some tests for this behaviour.

Is this critical for v5.0.0? If not, it might be best to put this in a patch release if it isn't ready to go.

**Comment by jemrobinson** [comment]
@JimMadge : I don't think it's essential and there are a few things needed here anyway (tests, use an enum). Mainly I think we should have *something* in the docs that explains how to deploy a tier 0/1/2/3/4 environment. Doesn't have to be this, but we should have some documentation instead in that case.

File: data_safe_haven/config/sre_config.py [hunk 1]
```diff
@@ -44,8 +44,34 @@ def from_remote_by_name(
         return cls.from_remote(context, filename=sre_config_name(sre_name))
 
     @classmethod
-    def template(cls: type[Self]) -> SREConfig:
+    def template(cls: type[Self], tier: int | None = None) -> SREConfig:
         """Create SREConfig without validation to allow "replace me" prompts."""
+        # Set tier-dependent defaults
+        if tier == 0:  # noqa: PLR2004
```
Comment (jemrobinson) [inline data_safe_haven/config/sre_config.py]:
```suggestion
        if tier == 0:
```

File: data_safe_haven/config/sre_config.py [hunk 2]
```diff
@@ -44,8 +44,34 @@ def from_remote_by_name(
...
+            remote_desktop_allow_copy = True
+            remote_desktop_allow_paste = True
+            software_packages = SoftwarePackageCategory.ANY
+        elif tier == 1:  # noqa: PLR2004
```
Comment (jemrobinson) [inline data_safe_haven/config/sre_config.py]:
```suggestion
        elif tier == 1:
```

**Review by JimMadge — APPROVED** [review]

**Pull request merged**